### PR TITLE
Wvr manager prepare land

### PR DESCRIPTION
**Pull request by tiagovignatti**
Hi back! This continues my changes in preparation for the WebXR shared buffer transport work coming next.

Just like my previous PR, there are no functional changes in this PR. I've tested this using Meta Quest 2 on WebXR Samples and other immersive experiences from Wolvic init page. PTAL.

### CODE REVIEW

**Comment by zakharvoit** [comment]
nit: I think some files are not formatted with Chromium style (I know that we didn't follow that too strictly in the past). I would suggest running "clang-format -i <filename>" at least for the new files but ideally for all changed files unless it generates too much noise.

**Review by zakharvoit — APPROVED** [review]
Approved but added a nit

File: wolvic/browser/vr/wvr_api.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+// found in the LICENSE file.
+
+#include "wolvic/browser/vr/wvr_api.h"
+
+#include "base/android/jni_string.h"
+#include "wolvic/jni_headers/VRManager_jni.h"
+
+namespace wolvic {
+
+WvrApi::WvrApi() {
```
Comment (svillar) [inline wolvic/browser/vr/wvr_api.cc:12]: I don't like the name either, but let's forget about it at the moment.

File: wolvic/browser/vr/wvr_api.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+namespace wolvic {
+
+class WvrApi {
+ public:
+  WvrApi();
+
+  WvrApi(const WvrApi&) = delete;
+  WvrApi& operator=(const WvrApi&) = delete;
+
+  ~WvrApi();
```
Comment (svillar) [inline wolvic/browser/vr/wvr_api.h]: just do `= default` here and you can get rid of the destructor in the cc file

File: wolvic/browser/vr/wvr_api.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include "wolvic/browser/vr/moz_external_vr.h"
+
+namespace wolvic {
+
+class WvrApi {
+ public:
+  WvrApi();
+
+  WvrApi(const WvrApi&) = delete;
+  WvrApi& operator=(const WvrApi&) = delete;
```
Comment (svillar) [inline wolvic/browser/vr/wvr_api.h:21]: isn't there any macro in chromium to delete those? Anyway not very important

File: wolvic/browser/vr/wvr_manager.h [hunk 4]
```diff
@@ -5,34 +5,29 @@
...
-#include "ui/gfx/geometry/rect_f.h"
-#include "ui/gfx/geometry/size.h"
-#include "ui/gl/gl_helper.h"
-#include "wolvic/browser/vr/moz_external_vr.h"
 #include "wolvic/browser/vr/wvr_graphics_delegate.h"
 
+namespace device {
+class MailboxToSurfaceBridge;
+}
+
```
Comment (svillar) [inline wolvic/browser/vr/wvr_manager.h:24]: Unrelated looks like

**Review by svillar — CHANGES_REQUESTED** [review]

File: wolvic/browser/vr/wvr_manager.h (hunk 4 above)
Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.h:24]: this is intentional. Please refer to forward-declare nested classes: 
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++.md#forward-declarations-vs_includes
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-dos-and-donts.md#minimize-code-in-headers

File: wolvic/browser/vr/wvr_api.h (hunk 3 above)
Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_api.h:21]: I don't think so. Grepping for `& operator=(const ` shows thousands of similar occurrences.

**Comment by tiagovignatti** [comment]
Unfortunately I couldn't find a better way to rebase this branch (`wvr-manager-prepare-land`) on top of the newest `wolvic_experimental`, so I've rebased locally and forced a push my changes into `wvr-manager-prepare-land`.

Regardless, I believe I've addressed all the review comments now. I'll proceed and merge. Thank you!

File: wolvic/browser/vr/wvr_manager.h (hunk 4 above)
Comment (svillar) [reply wolvic/browser/vr/wvr_manager.h:24]: I know you're forward declaring, but as far as I saw you are still not using it

Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.h:24]: ah, I know what made the confusion: I actually have cleaned up some headers being included in this file without explicitly mentioning it. You will see a`- #include "components/webxr/mailbox_to_surface_bridge_impl.h"
` but GitHub doesn't actually show the `device::MailboxToSurfaceBridge` class being used bellow in `WvrManager` (it's used to define a pointer called `mailbox_bridge_`).

Sorry about not being clear about that.

**Pull request merged**